sklearn/linear_model/logistic.py

```diff
@@ -1220,15 +1240,18 @@ def fit(self, X, y, sample_weight=None):
         if self.multi_class == 'multinomial':
             classes_ = [None]
             warm_start_coef = [warm_start_coef]
-
+        else:
+            # Benchmark
+            self.coef_ = np.zeros((n_classes, n_features))
         if warm_start_coef is None:
             warm_start_coef = [None] * n_classes
 
         path_func = delayed(logistic_regression_path)
 
         # The SAG solver releases the GIL so it's more efficient to use
         # threads for this solver.
-        backend = 'threading' if self.solver == 'sag' else 'multiprocessing'
+        backend = 'threading' if self.solver in ['sag', 'saga']\
+            else 'multiprocessing'
```

Comment:
Please do no use line break escaping. If the ternary conditional expression does not fit on a single line, just use a traditional multiline conditional. It's more readable:

```python

if self.solver in ['sag', 'saga']:
    backend = 'threading'
else:
    backend = 'multiprocessing'
```

BTW @TomDLT did you try to use the threading backend with the other solvers? Don't they release the GIL?

For liblinear we should release the GIL in our bindings if this is not already the case. For the scipy solvers I am pretty sure they do it already. If not we should report this as an issue on their tracker.